test_container/tests/test/pandas/all/dataframe_memory_leak.py

```diff
@@ -0,0 +1,228 @@
+from typing import List, Tuple, Union
+
+
+class PandasDataFrameMemoryLeakTest(udf.TestCase):
+    def setUp(self):
+        self.maxDiff=None
+
+        self.query('CREATE SCHEMA FN2', ignore_errors=True)
+        self.query('OPEN SCHEMA FN2', ignore_errors=True)
+
+        self.create_col_defs = [
+            ('C0','INT IDENTITY'),
+            ('C1','Decimal(2,0)'), 
+            ('C2','Decimal(4,0)'),
+            ('C3','Decimal(8,0)'),
+            ('C4','Decimal(16,0)'),
+            ('C5','Decimal(36,0)'),
+            ('C6','DOUBLE'),
+            ('C7','BOOLEAN'),
+            ('C8','VARCHAR(500)'),
+            ('C9','CHAR(10)'),
+            ('C10','DATE'),
+            ('C11','TIMESTAMP')
+            ]
+        self.create_col_defs_str = ','.join(
+                '%s %s'%(name,type_decl) 
+                for name, type_decl 
+                in self.create_col_defs
+                )
+        self.col_defs = self.create_col_defs[1:]
+        self.col_defs_str = ','.join(
+                '%s %s'%(name,type_decl) 
+                for name, type_decl 
+                in self.col_defs
+                )
+        self.col_names = [name for name, type_decl in self.col_defs]
+        self.col_names_str = ','.join(self.col_names)
+
+        self.col_tuple = (
+                Decimal('1'), 
+                Decimal('1234'), 
+                Decimal('12345678'), 
+                Decimal('1234567890123456'), 
+                Decimal('123456789012345678901234567890123456'), 
+                12345.6789, 
+                True, 
+                'abcdefghij', 
+                'abcdefgh  ', 
+                date(2018, 10, 12), 
+                datetime(2018, 10, 12, 12, 15, 30, 123000)
+                )
+        self.create_table_1()
+
+
+    def create_table(self,table_name,create_col_defs_str):
+        create_table_sql='CREATE TABLE %s (%s)' % (table_name,create_col_defs_str)
+        print("Create Table Statement %s"%create_table_sql)
+        self.query(create_table_sql)
+
+    def create_table_1(self):
+        self.create_table("TEST1",self.create_col_defs_str)
+        self.import_via_insert("TEST1",[self.col_tuple],column_names=self.col_names)
+        num_inserts = 17 # => ~128 K rows
+        for i in range(num_inserts):
+            insert_sql = 'INSERT INTO TEST1 (%s) SELECT %s FROM TEST1' % (self.col_names_str, self.col_names_str)
+            print("Insert Statement %s"%insert_sql)
+            self.query(insert_sql)
+        self.num_rows = 2**num_inserts
+
+    def test_dataframe_scalar_emits(self):
+        """
+        This test checks that the largest memory block of a tracemalloc snapshot diff is not larger than 100KB, where
+        the memory block snapshots are retrieved during the first/last invocation of the scalar UDF,
+        but after the emit().
+        Reasoning for 100KB is that the number of rows is > 100K, so if there was 1 Byte leaking during every execution,
+        it would be found here.
```

Comment:
This means we could still leak memory during the first emit

Reply:
If I retrieve the snapshot before the first get_dataframe/emit, all the python packages (numpy, pandas, etc.) are not loaded yet. They pollute the diff. I'am interested in this test if the memory usage increases by iterative calls to emit,

Reply:
To avoid the pollution, I would load pandas and numpy manually and move the first take snapshot before the first get_dataframe

Reply:
As discussed offline, let's keep it like this for now. The first invocation of emit will allocate a lot of memory, this makes it more difficult to find a reasonable number for the limit and thus finding a memory leak when calling emit() many time....